project_development_tips.md

```diff
@@ -0,0 +1,119 @@
+    - Even if the project is relatively easy, the owner should be someone who understands Cadence well and who keeps in touch with what is happening in the Cadence world. Examples:
+        - Responsible project maintainers should contribute to discussions about important proposals (new cadence features, standard smart contracts, metadata, etc) and generally be aware about evolving best practices and anti-pattern understandings.
+        - They should regularly re-evaluate the code of the project to see if it needs to be updated to conform to standards and new versions of Cadence.
+        - The best projects should be active in the development of the protocol and language.
+    - The owner should also understand how to sign transactions with the CLI to upgrade the smart contract, run admin transactions, deploy new contracts, etc. 
+- The owner should identify a secondary owner who doesn’t hold any direct responsibility, but still has a clear understanding of the code and requirements so they can give good feedback, perform effective reviews, and make changes where needed. In case the owner is not available or leaves the project, this person can also take over easily so there is never a time when there is nobody who understands the smart contracts well.
+- If there isn’t already an open source repo, the owner should create a public GitHub repo from [the open source template](https://github.com/onflow/open-source-template) and make sure all of this is set up with some initial documentation for what the repo is for before any code is written.
+- The owner should create a design document that lays out the intended high level design and architecture of the smart contract and some basic user stories, as well as any questions that still need to be answered about it.
+    - Diagrams should be made where applicable describing state machines, user flows, etc
+    - This document should be shared in an issue in the open source repo where the contracts or features are being developed.
+- If the owner has to move to another team, a new owner on the project’s team should be identified and ramped up immediately so that there is always an owner who is intimately involved with the project and is aware of all the details.
+- Keys for the contract and/or admin accounts should be stored in a secure key store such as google KMS or something similar, and the owner should have access to these along with the secondary owner and some other engineering manager. Keys should never be in any file in the public repo.
+- Project owners need to always keep [composability and extensibility](https://www.notion.so/Smart-Contract-Composability-be544f21a2e84596affad73952b21d56) in mind while designing, developing, and documenting their projects.
```

Comment:
I guess this link is internal? Should we make a public version of this too?

Reply:
yes, that is internal. I'll covert that to a doc here too